### PR TITLE
ref(ui): Simplify badges implementation

### PR DESCRIPTION
**Pull request by evanpurkhiser**
- The UserBadge no longer has a bunch of custom styling and uses the
  existing styles

- The MemberBadge delegates to the User badge now

### CODE REVIEW

File: static/app/components/idBadge/memberBadge.tsx [hunk 1]
```diff
@@ -32,82 +25,16 @@ function getMemberUser(member: Member): AvatarUser {
...
-  color: ${p => p.theme.gray300};
-  ${p => p.theme.overflowEllipsis};
-`;
-
-interface NameProps {
-  hideEmail: boolean;
-  to: LinkProps['to'];
-  useLink: boolean;
-  children?: React.ReactNode;
+  return <UserBadge to={membersUrl} user={user} {...props} />;
```
Comment (ryan953) [inline static/app/components/idBadge/memberBadge.tsx:37]: this is amazing

File: static/app/components/idBadge/baseBadge.tsx [hunk 2]
```diff
@@ -57,19 +61,13 @@ const BaseBadge = memo(
   )
 );
 
-export default BaseBadge;
-
 const Wrapper = styled('div')`
   display: flex;
+  gap: ${space(1)};
```
Comment (ryan953) [inline static/app/components/idBadge/baseBadge.tsx:66]: Idk if this is properly replacing the margin-right that StyledAvatar had before.

The problem I'm seeing is that the replay breadcrumbItem has no spacing now:
SCR-20240416-jpqv

Comment (ryan953) [reply static/app/components/idBadge/baseBadge.tsx:66]: waiting to see the combo with https://github.com/getsentry/sentry/pull/69009, will return to look again!

Comment (evanpurkhiser) [reply static/app/components/idBadge/baseBadge.tsx:66]: this will have a merge conflict once #69009 merges

**Review by ryan953 — APPROVED** [review]

**Pull request merged**